### PR TITLE
Rev refactor

### LINKED ISSUE #32: Rev task should work in dev as well
In helper-rev, we have a fallback to `dist` if the manifest can't be found.

This should actually be conditional and if in local dev, look in the `dev` folder.

### LINKED ISSUE #34: Remove JS bundling/concatenation
Minification and gzip are still good, but bundling everything into one big JS file is actually an [antipattern](http://stackoverflow.com/questions/30861591/why-bundle-optimizations-are-no-longer-a-concern-in-http-2)

[WP Engine](http://stackoverflow.com/questions/30861591/why-bundle-optimizations-are-no-longer-a-concern-in-http-2) has HTTP2 (if you have SSL).

I propose something like this, which would let you still have a single version of libs, but make each page only load what it needs.

`template-something.twig`:

```
{% load_footer_scripts = [1, 2] %}
```

`layout.twig`:

```
{% if 'slick' in load_footer_scripts %}
    <script src="{{ function("rev", "vendor/slick-carousel/slick/slick.js") }}"></script>
{% endif %}
```

Individual pages could either load JS inline using `{% block footer_scripts %}`, or make a call to something in `js/pages/something-complicated.js

### LINKED ISSUE #43: Prevent Gulp useref from copying layout .twig to dev/dist
The useref task duplicated layout.twig to dev/dist. Sometimes when searching in sublime for layout.twig, I'll inadvertently edit this file.

### LINKED ISSUE #59: Update gulp, try node 6
Let's update gulp plugin versions, and see if node 6 works

### PR DESCRIPTION
**Pull request by ccorda**
Closes #32 
Closes #34 
Closes #43 
Closes #59

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -102,7 +102,8 @@ gulp.task('styles', function() {
 
 gulp.task('scripts', function() {
     var assets = $.useref({
-        searchPath: './'
+        searchPath: config.assets,
+        noconcat: true,
```
Comment (kylehotchkiss) [inline gulpfile.js]: Does `noconcat` minify the files in-place and not combine them?

File: wp-content/themes/timber/views/layout.twig [hunk 2]
```diff
@@ -85,26 +80,31 @@
...
-            <script src="/wp-content/themes/timber/assets/vendor/twbs-bootstrap-sass/assets/javascripts/bootstrap/dropdown.js"></script>
-            <script src="/wp-content/themes/timber/assets/vendor/quickshare/dist/quickshare.js"></script>
-            <script src="/wp-content/themes/timber/assets/vendor/matchHeight/jquery.matchHeight.js"></script>
-
-            <script src="/wp-content/themes/timber/assets/js/console-safe.js"></script>
-            <script src="/wp-content/themes/timber/assets/js/analytics-sourcing.js"></script>
-            <script src="/wp-content/themes/timber/assets/js/main.js"></script>
+            <script src="vendor/query-string/query-string.js"></script>
+            <script src="vendor/jquery-smooth-scroll/jquery.smooth-scroll.js"></script>
+            <script src="vendor/cookies-js/dist/cookies.js"></script>
```
Comment (kylehotchkiss) [inline wp-content/themes/timber/views/layout.twig]: Should we move towards using localstorage instead?

Comment (ccorda) [inline wp-content/themes/timber/views/layout.twig]: In general, yes. However three pros of cookies here:

1) built in expiration
2) crossdomain, which is critical if CRM is on a different subdomain
3) BSD source and subsource have to be cookies in order to work

File: gulpfile.js (hunk 1 above)
Comment (ccorda) [inline gulpfile.js]: The uglify line is what does minification, and that stays the same. But the goal here is not to concatenate, as that is now a performance anti-pattern in the age of http2. (see #34)

**Comment by ccorda** [comment]
Thanks Kyle. Yes, confirmed that version 6 builds no problem on travis, so moving forward with this.

**Pull request merged**